packages/cyberstorm/src/components/Pagination.tsx

```diff
@@ -0,0 +1,140 @@
+import { faArrowLeft, faArrowRight } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import React, { Dispatch, SetStateAction } from "react";
+import { range } from "../utils";
+import styles from "./componentStyles/Pagination.module.css";
+import { PaginationButton } from "./PaginationButton";
+
+export interface PaginationProps {
+  currentPage: number;
+  onPageChange: Dispatch<SetStateAction<number>>;
+  totalCount: number;
+  pageSize: number;
+  disabled?: boolean;
+  siblingCount?: number;
+}
+
+const DOTS = "…";
+
+const getButtonLabels = (
+  totalPageCount: number,
+  currentPage: number,
+  siblingCount = 1
+) => {
+  // maxButtons is determined as siblingCount + firstPage + lastPage + currentPage + 2*DOTS
+  const maxButtons = siblingCount + 5;
+
+  // Show a button for each page if there's just a few of them.
+  if (maxButtons >= totalPageCount) {
+    return range(1, totalPageCount);
+  }
+
+  const prevPageNumber = Math.max(currentPage - siblingCount, 1);
```

Comment:
With all the other refactorings done here, I'm not sure these names are very good anymore. E.g. with `siblingCount` 3 and `currentPage` 3, `prevPageNumber` will be 1 and `nextPageNumber` 6, while one would probably expect 2 and 4. Would `leftmostSibling` and `rightmostSibling` make sense?

Reply:
This sense shall be made to be true!